src/Components/ArtsyShippingEstimate.tsx

```diff
@@ -0,0 +1,292 @@
+  "Mixed Media": "mixed_media_framed",
+  "Design/Decorative Art": "other_decorative_arts",
+  Installation: "other_art",
+  "Textile Arts": "tapestry",
+  "Books and Portfolios": "book",
+  "Ephemera or Merchandise": "memorabilia", // temporary fix while arta builds out support for self ship types
+  Other: "other_art",
+} as const
+
+const UNFRAMED_CATEGORY_MAP = {
+  Photography: "photograph_unframed",
+  Painting: "painting_unframed",
+  Print: "work_on_paper_unframed",
+  Posters: "work_on_paper_unframed",
+  "Drawing, Collage or other Work on Paper": "work_on_paper_unframed",
+  "Mixed Media": "mixed_media_unframed",
+  Sculpture: "sculpture",
+  "Design/Decorative Art": "other_decorative_arts",
+  Installation: "other_art",
+  "Textile Arts": "tapestry",
+  "Books and Portfolios": "book",
+  "Ephemera or Merchandise": "memorabilia", // temporary fix while arta builds out support for self ship types
+  Other: "other_art",
+} as const
+
+interface ShippableArtwork {
+  shippingOrigin: string
+  location: {
+    country: string
+    postalCode?: string
+    city?: string
+  }
+
+  priceCurrency: SupportedCurrency
+  priceListed: { major: number }
+  isFramed: boolean
+  category:
+    | keyof typeof FRAMED_CATEGORY_MAP
+    | keyof typeof UNFRAMED_CATEGORY_MAP
+  framedMetric?: string
+  framedHeight?: number
+  framedDiameter?: number
+  framedWidth?: number
+  framedDepth?: number
+  heightCm?: number
+  diameterCm?: number
+  widthCm?: number
+  depthCm?: number
+}
+
+type ArtaObjectDimensions = Pick<
+  ArtaObject,
+  "height" | "width" | "unit_of_measurement" | "depth"
+>
+const artworkDimensions = (
+  artwork: ShippableArtwork,
+): ArtaObjectDimensions | null => {
+  if (artwork.isFramed && !!artwork.framedMetric) {
+    if (!!(!!artwork.framedHeight && !!artwork.framedWidth)) {
+      return {
+        unit_of_measurement: artwork.framedMetric,
+        height: artwork.framedHeight,
+        width: artwork.framedWidth,
+        depth: artwork.framedDepth,
+      }
+    }
+    if (!!artwork.framedDiameter) {
+      return {
+        unit_of_measurement: artwork.framedMetric,
+        height: artwork.framedDiameter,
+        width: artwork.framedDiameter,
+        depth: artwork.framedDepth,
+      }
+    }
+  } else if (!!artwork.heightCm && !!artwork.widthCm) {
+    return {
+      unit_of_measurement: "cm",
+      height: artwork.heightCm,
+      width: artwork.widthCm,
+      depth: artwork.depthCm,
+    }
+  } else if (!!artwork.diameterCm) {
+    return {
+      unit_of_measurement: "cm",
+      height: artwork.diameterCm,
+      width: artwork.diameterCm,
+      depth: artwork.depthCm,
+    }
+  }
+  return null
+}
+
+type ArtworkValue = Pick<ArtaObject, "value" | "value_currency">
+const artworkValue = (artwork: ShippableArtwork): ArtworkValue | null => {
+  if (!!artwork.heightCm && !!artwork.widthCm) {
+    return {
+      value: artwork.priceListed?.major || 500, // TODO: get a real value
```

Comment:
I think `price_listed` is the backend price that is [not exposed publicly](https://github.com/artsy/gravity/blob/f99e2848373c795f1b06d6eacbba219ca4855729/app/models/domain/artwork.rb#L1379) so I guess users will not be able to see this. Also, do we aim to support artworks displayed as price range, etc? (sorry if I missed the product requirements)

Reply:
Good point @starsirius! I believe there was not a discussion about excluding range works. And I forgot about this whole 'price being private' for MO sometimes. Think taking midpoint makes sense here but def something to discuss.

Reply:
Perhaps [this is the right field](https://github.com/artsy/metaphysics/blob/4d5fab5395d127f5d4de0470b62a80e219455ee3/src/schema/v2/artwork/index.ts#L1274) for public facing pricing. Confusing indeed!